cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,360 @@
+            to run MMR
+            n: number of words per topic (Default: 30)
+        Returns:
+            top_n_words: The top n words per topic
+            topic_names: Dictionary containing key as Topic ID and value
+            as top 4 words from that topic cluster
+        """
+
+        words = count.get_feature_names().to_arrow().to_pylist()
+        labels = sorted(docs_per_topics_topics.to_arrow().to_pylist())
+        indices = top_n_idx_sparse(tf_idf, n)
+        indices = indices.get()
+
+        top_n_words = {}
+        for i, label in enumerate(labels):
+            list_labels = []
+            indices_row = indices[i]
+            for idx in indices_row:
+                if not math.isnan(idx):
+                    idx = int(idx)
+                    if idx and tf_idf[i, idx] > 0:
+                        list_labels.append((words[idx], tf_idf[i, idx]))
+                    else:
+                        list_labels.append(("", 0.00001))
+            top_n_words[label] = list_labels[::-1]
+
+        if mmr_flag:
+            for topic, topic_words in top_n_words.items():
+                words_arr = [word[0] for word in topic_words]
+                words = cudf.Series(words_arr, name="Document")
+                word_embeddings = create_embeddings(words)
+                topic_embedding = create_embeddings(
+                    cudf.Series(" ".join(words_arr))
+                ).reshape(1, -1)
+                topic_words = mmr(
+                    topic_embedding,
+                    word_embeddings,
+                    words,
+                    top_n=n,
+                    diversity=0
+                )
+                top_n_words[topic] = [
+                    (word, value)
+                    for word, value in top_n_words[topic]
+                    if word in topic_words
+                ]
+
+        top_n_words = {
+            label: values[:n] for label, values in top_n_words.items()
+        }
+
+        topic_names = {
+            key: f"{key}_" + "_".join([word[0] for word in values[:4]])
+            for key, values in top_n_words.items()
+        }
+        return top_n_words, topic_names
+
+    def extract_topic_sizes(self, df):
+        """Calculate the topic sizes
+        Arguments:
+            documents: dataframe with documents and their corresponding IDs
+            and added Topics
+
+        Returns:
+            topic_sizes: DataFrame containing topic cluster sizes
+        """
+        topic_sizes = (
+            df.groupby(["Topic"])
+            .Document.count()
+            .reset_index()
+            .rename({"Topic": "Topic", "Document": "Count"}, axis="columns")
+            .sort_values("Count", ascending=False)
+        )
+        return topic_sizes
+
+    def fit_transform(self, data):
+        """Fit the models on a collection of documents, generate topics, and return
+        the docs with topics
+        Arguments:
+            documents: A list of documents to fit on
```

Comment:
Arguments and docstring dont line  up, documents .  I suggest changing it to `documents`